src/ui/layout/tutorial.md

```diff
@@ -244,219 +393,287 @@ class MyApp extends StatelessWidget {
-    'half-hour walk through pastures and pine forest, leads you to the '
-    'lake, which warms to 20 degrees Celsius in the summer. Activities '
-    'enjoyed here include rowing, and riding the summer toboggan run.',
-    softWrap: true,
-  ),
-);
+class TextSection extends StatelessWidget {
+  const TextSection({
+    super.key,
+    required this.description,
+  });
+
+  final String description;
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.all(32),
+      child: Text(
+        description,
+        softWrap: true,
+      ),
+    );
+  }
+}
 ```
 
-By setting `softwrap` to true, text lines will fill the column width before
+By setting `softwrap` to `true`, text lines fill the column width before
 wrapping at a word boundary.
 
-Add the text section to the body:
+### Update the app to display the text section
 
-<?code-excerpt path-base="layout/lakes"?>
-<?code-excerpt "{step3,step4}/lib/main.dart" from="return MaterialApp"?>
+Insert the `TextSection` call into the `body` property.
+This call needs `description` variable set to the text of the
+location description.
+
+<?code-excerpt "step{3,4}/lib/main.dart (addWidget)" title?>
 ```diff
---- step3/lib/main.dart
-+++ step4/lib/main.dart
-@@ -59,3 +72,3 @@
-     return MaterialApp(
-       title: 'Flutter layout demo',
-       home: Scaffold(
-@@ -66,6 +79,7 @@
-           children: [
-             titleSection,
-             buttonSection,
-+            textSection,
-           ],
-         ),
+--- step3/lib/main.dart (addWidget)
++++ step4/lib/main.dart (addWidget)
+@@ -6,6 +6,16 @@
+         location: 'Kandersteg, Switzerland',
        ),
+       ButtonSection(),
++      TextSection(
++        description:
++            'Lake Oeschinen lies at the foot of the Blüemlisalp in the '
++            'Bernese Alps. Situated 1,578 meters above sea level, it '
++            'is one of the larger Alpine Lakes. A gondola ride from '
++            'Kandersteg, followed by a half-hour walk through pastures '
++            'and pine forest, leads you to the lake, which warms to 20 '
++            'degrees Celsius in the summer. Activities enjoyed here '
++            'include rowing, and riding the summer toboggan run.',
++      ),
+     ],
+   ),
+ ),
 ```
 
-## Step 5: Implement the image section
+## Add the Image section
 
-Three of the four column elements are now complete,
-leaving only the image. Add the image file to the example:
+In this section, add the image file to complete your layout.
 
-* Create an `images` directory at the top of the project.
-* Add [`lake.jpg`][].
+### Configure your app to use supplied images
 
-{{site.alert.info}}
-  Note that `wget` doesn't work for saving this binary file.
-  The original image is [available online][] under a
-  Creative Commons license, but it's large and slow to fetch.
-{{site.alert.end}}
+In this section, configure your app to reference images in your Flutter project.
```

Comment:
Rather than stating what they will do so formally, consider opening with details on what they are going to do now and why its important. Outside of something at the beginning, I feel readers glaze over "In this section..." and it can get a bit repetitive.

Could start the intro with something like:

```
To configure your app to use custom images...
```